test/unit/shared/redis-connection.test.ts

```diff
@@ -138,30 +185,33 @@ describe('RedisConnection', () => {
     } catch (error) {
       expect(error).toEqual(new Error('emitted'))
       expect(config.logger.push).toHaveBeenCalledWith({ err: new Error('emitted') })
-      expect(config.logger.error).toHaveBeenCalledWith('Error from REDIS client')
+      expect(config.logger.error).toHaveBeenCalledWith('createClient: Error from REDIS client')
     }
   })
 
   it('should protected against multiple rejection when ready but log errors', (done): void => {
     const createClientSpy = jest.spyOn(Redis, 'createClient')
     createClientSpy.mockImplementationOnce(() => (
       {
+        quit: jest.fn(),
```

Comment:
Should there be a another test that simulates a reconnection failure and ensures that `client.quit()` is called?

Reply:
@lewisdaly good point! Tests improved! Please take a look again